presto-spark-base/src/main/java/com/facebook/presto/spark/util/PrestoSparkStatsCollectionUtils.java

```diff
@@ -0,0 +1,157 @@
+        Long metricVal = getMetricValue(metric, isSparkUnitMs);
+        incSparkInternalAccumulator(sparkInternalAccumulatorKey, prestoKey, metricVal);
+    }
+
+    static String getSparkInternalAccumulatorKey(final String prestoKey)
+    {
+        if (prestoKey.contains(SPARK_INTERNAL_ACCUMULATOR_PREFIX)) {
+            if (prestoKey.contains(PRESTO_NATIVE_OPERATOR_STATS_SEP)) {
+                String[] strs = prestoKey.split(PRESTO_NATIVE_OPERATOR_STATS_SEP);
+                if (strs == null || strs.length < 2) {
+                    log.info("Fail to build spark internal key for %s format not supported", prestoKey);
+                    return "";
+                }
+                return strs[1];
+            }
+        }
+        String[] strs = prestoKey.split("\\.");
+        if (strs == null || strs.length < 2) {
+            log.info("Fail to build spark internal key for %s format not supported", prestoKey);
+            return "";
+        }
+        String prestoNewKey = String.format("%1$s%2$s", strs[0], strs[strs.length - 1]);
+        if (prestoNewKey.contains("_")) {
+            prestoNewKey = CaseUtils.toCamelCase(prestoKey, false, '_');
+        }
+        return String.format("%1$s%2$s%3$s", SPARK_INTERNAL_ACCUMULATOR_PREFIX,
+                PRESTO_NATIVE_OPERATOR_STATS_PREFIX, prestoNewKey);
+    }
+
+    static Set<RuntimeStats> collectRuntimeStats(TaskInfo taskInfo)
+    {
+        Set<RuntimeStats> stats = new HashSet<>();
+        if (taskInfo.getStats() == null) {
+            return stats;
+        }
+        for (PipelineStats pipelineStats : taskInfo.getStats().getPipelines()) {
+            if (pipelineStats != null) {
+                for (OperatorStats operatorStats : pipelineStats.getOperatorSummaries()) {
+                    if (operatorStats != null) {
+                        if (operatorStats.getOperatorType().equals("NativeExecutionOperator")) {
+                            NativeExecutionInfo nativeExecutionInfo = (NativeExecutionInfo) operatorStats.getInfo();
+                            if (nativeExecutionInfo != null) {
+                                for (TaskStats taskStats : nativeExecutionInfo.getTaskStats()) {
+                                    if (taskStats != null) {
+                                        RuntimeStats runtimeStat = taskStats.getRuntimeStats();
+                                        stats.add(runtimeStat);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return stats;
+    }
+
+    static Long getMetricValue(RuntimeMetric metric, boolean isSparkUnitMs)
+    {
+        BigDecimal sum = BigDecimal.valueOf(metric.getSum());
+        if (metric.getUnit().equals(RuntimeUnit.NANO) && isSparkUnitMs) {
+            sum = sum.divide(BigDecimal.valueOf(1000000), BigDecimal.ROUND_DOWN);
```

Comment:
we can use the built-in lib to convert unit btw ns to ms:` TimeUnit.NANOSECONDS.toMillis(nanoSec);`